text/testing/mock.md

```diff
@@ -36,27 +36,56 @@ impl Trait for TestRuntime {
 Next, we create a new type that wraps the mock `TestRuntime` in the pallet's `Module`.
 
 ```rust, ignore
-pub type HelloSubstrate = Module<TestRuntime>;
+pub type TestPallet = Module<TestRuntime>;
 ```
 
 It may be helpful to read this as type aliasing our configured mock runtime to work with the pallet's `Module`, which is what is ultimately being tested.
 
 ## `impl system::Trait`
 
-In many cases, the pallet's `Trait` inherits `system::Trait` like
+In many cases, the pallet's `Trait` inherits `system::Trait` like:
 
 ```rust, ignore
 pub trait Trait: system::Trait {
 	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
 }
 ```
 
-The mock runtime must inherit and define the `system::Trait` associated types (*[remember](https://substrate.dev/recipes/traits/index.html)*). To do so, `impl` the `system::Trait` for `TestRuntime` with a few types imported from other crates,
+Before we can create a mock runtime that take our pallet to run tests, we first need to create the
+outer environment for the runtime, as follows:
 
 ```rust, ignore
 use support::{impl_outer_event, impl_outer_origin, parameter_types};
 use runtime_primitives::{Perbill, traits::{IdentityLookup, BlakeTwo256}, testing::Header};
+use runtime_io;
+use primitives::{H256};
+
+// We define the outer `Origin` enum, and `Event` enums.
+// You may not aware these enums are created when writing the pallet, it is because
+//   they are created through the macro `decl_event`, `decl_module` that we use.
```

Comment:
The _outer_ enums like `outer_origin`, `outer_event`, `outer_error` etc are created by `construct_runtime!`.

It's the inner enums that are constructed by macros like `decl_event!`.

Reply:
I put this at the top of the section on implementing traits.

Reply:
Indeed. Just checked with `cargo expand`.